### PR TITLE
Add a debug/nodbg component to the runtime unique configuration path.

### PR DESCRIPTION
**Pull request by gbtitus**
Whether or not the `CHPL_COMM_DEBUG` environment variable is set changes
what code is built in comm layers other than 'none', as well as changing
how GASNet is configured if we're using that.  However, although that
variable also affected the GASNet unique configuration path it didn't
affect the runtime one, so you could end up using a debug gasnet comm
layer with a non-debug GASNet build or vice-versa.  Or for multi-file
comm layers like ofi, you could end up mixing comm layer files built for
debug with comm layer files built for non-debug.  Here, resolve these
problems by adding a 'debug' or 'nodbg' component to the runtime unique
config path based on the setting of `CHPL_COMM_DEBUG`.

Note that because it changes the runtime unique configuration path this
will effectively invalidate existing runtime builds other than for comm=none,
and cause new ones to be done.

This resolves https://github.com/Cray/chapel-private/issues/626.

### CODE REVIEW

File: util/chplenv/printchplenv.py [hunk 1]
```diff
@@ -338,6 +338,9 @@ def printchplenv(contents, print_filters=None, print_format='pretty'):
         if env.name == 'CHPL_TARGET_CPU' and print_format == 'path':
             value = ENV_VALS['CHPL_RUNTIME_CPU']
         ret.append(print_var(env.name, value, shortname=env.shortname))
+        if env.name == 'CHPL_COMM' and print_format == 'path':
+            if value != 'none':
```
Comment (ronawho) [inline util/chplenv/printchplenv.py]: It's conceivable we could have debug options for comm=none code too, right?

**Comment by ronawho** [comment]
Can/should we use this to compute the path in third-party/gasnet/Makefile.include instead of hardcoding debug vs nodbg again?

I've also found it a little weird that we embed nodbg instead of just not having anything. My personal preference would be to have this show up in the path like `comm-gasnet{,-debug}`, but I'm not sure how much more work that is. What's here is better than what we have, so I'm fine going with this. I also wonder if we'll ever want debug flags for tasking/memory/whatever, but I'm also happy to save that for when we need it.

**Comment by gbtitus** [comment]
> Can/should we use this to compute the path in third-party/gasnet/Makefile.include instead of hardcoding debug vs nodbg again?

Not directly, because the comm layer component doesn't even occur in the GASNet unique config path.  It's implicit, thus no need to express it.  The current GASNet unique cfg path is `targetPlatform-targetCompiler-targetCPU-libPIC/substrate-XYZ/seg-XYZ/{debug,nodbg}`.  I could switch the leading component to be the default third-party unique config path `targetPlatform-targetArch-targetCPU-targetCompiler-libPIC` with an optional `-debug` (or `/debug`?), and leave the substrate and segment components as they are.  (I was actually surprised the `targetArch` wasn't already in the GASNet path.)

**Comment by ronawho** [comment]
Ah, I was thinking we could just replace `CHPL_GASNET_DEBUG` with `CHPL_MAKE_COMM_DEBUG`, but we're intentionally not exposing the variable like that.

**Comment by gbtitus** [comment]
@ronawho I think this is ready for a last look now.

**Pull request merged**